### PR TITLE
Con 5203: Fix renaming variant options

### PR DESCRIPTION
**Pull request by Christian-Rades**

### CODE REVIEW

File: Components/VariantConfigurator.php [hunk 1]
```diff
@@ -80,14 +81,19 @@ public function configureVariantAttributes(Product $product, Detail $detail)
                 $detailOptions->add($option);
             }
         }
+
         $this->manager->persist($configSet);
 
         $detail->setConfiguratorOptions($detailOptions);
         $this->manager->persist($detail);
         $this->manager->persist($article);
+        $this->manager->persist($article);
```
Comment (SebastianFranze) [inline Components/VariantConfigurator.php]: Warum zweimal persist?

File: Components/VariantConfigurator.php [hunk 2]
```diff
@@ -80,14 +81,19 @@ public function configureVariantAttributes(Product $product, Detail $detail)
...
+
+        $this->deleteUnusedOptions($product, $detailOptions, $configSet);
 
         $this->manager->flush();
 
+
```
Comment (SebastianFranze) [inline Components/VariantConfigurator.php]: Zwei leere Zeilen

File: Components/VariantConfigurator.php [hunk 3]
```diff
@@ -98,6 +104,53 @@ public function configureVariantAttributes(Product $product, Detail $detail)
...
+    {
+        /** @var Option $detailOption */
+        foreach ($detailOptions as $detailOption) {
+            if (!in_array($detailOption->getName(), $product->variant)) {
+
+                $detailOptions->removeElement($detailOption);
+
+                $configSet->getOptions()->removeElement($detailOption);
+
+                $this->manager->flush();
```
Comment (SebastianFranze) [inline Components/VariantConfigurator.php]: muss flush in jedem Schleifendurchlauf?

Comment (Christian-Rades) [reply Components/VariantConfigurator.php]: Leider ja :(

File: Components/ProductToShop.php [hunk 4]
```diff
@@ -445,6 +449,25 @@ private function deleteDetail(DetailModel $detailModel)
...
+    {
+        $configSet = $article->getConfiguratorSet();
+        $options = $detailModel->getConfiguratorOptions();
+        if($options && $configSet) {
+            $optionCollection = $configSet->getOptions();
+            /** @var Option $option */
+            foreach ($options as $option) {
+                $optionCollection->removeElement($option);
+            }
+            $configSet->setOptions($options);
```
Comment (keulinho) [inline Components/ProductToShop.php]: does `setOptions()` append or does it overwrite the list of options?
i think it overrides and then the whole foreach is not necessary

File: Components/VariantConfigurator.php [hunk 5]
```diff
@@ -98,6 +102,53 @@ public function configureVariantAttributes(Product $product, Detail $detail)
...
+    /**
+     * @param $detailOption
+     * @return bool
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    private function isOptionUsed(Option $detailOption)
+    {
+        $connection = $this->manager->getConnection();
+        $query = $connection->prepare(
+            'SELECT option_id FROM s_article_configurator_set_option_relations WHERE option_id = :optionId UNION
```
Comment (keulinho) [inline Components/VariantConfigurator.php]: image
`UNION`

**Pull request merged**